utils/python/CIME/test_scheduler.py

```diff
@@ -148,13 +148,17 @@ def __init__(self, test_names, test_data=None,
 
         # Setup phases
         self._phases = list(PHASES)
+        # These will mark the -no phase as PENDING in TestStatus 
         if self._no_setup:
-            self._phases.remove(SETUP_PHASE)
+            i = self._phases.index(SETUP_PHASE)
```

Comment:
I'm not sure I see what you mean - are you saying if both are True?   
this if block could be

if self._no_setup:
elif self._no_build:
elif self._no_run:

would that resolve the concern?